### PR TITLE
Added ValidatableCommandClause POC

### PR DESCRIPTION
**Pull request by amorton**
**What this PR does**:
Encapsulates behaviour for clauses that can be validated and better supports schema object types.

This is a POC for adding this, just to the level needed to make table code work. 

I will create a ticket to centralize all of the validation, based on this approach, and add here. 

See https://github.com/stargate/data-api/issues/1310 

NOTE : THIS PR DOES NOT CLOSE THE ISSUE ABOVE ^^^^

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [*] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [*] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/ValidatableCommandClause.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+   * methods on the interface directly.
+   *
+   * @param commandContext The context the command is running against, including the {@link
+   *     SchemaObject}
+   * @param validatable An object that implements {@link ValidatableCommandClause}, may be null
+   * @param <T> Type of the {@link SchemaObject}
+   */
+  static <T extends SchemaObject> void maybeValidate(
+      CommandContext<T> commandContext, ValidatableCommandClause validatable) {
+    if (validatable == null) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/ValidatableCommandClause.java:34]: Is `null` legal? Assumption being that missing Clause is always valid thing?

**Review by tatu-at-datastax — APPROVED** [review]
Ok, I have some questions about the big picture, but details are fine. We can go back and consider this -- approving since it's a pre-req for other changes.

**Comment by Yuqi-Du** [comment]
Reviewed, approach is clear. 
+1 that we may need to make some exceptions, since not every validation is easy to do or good to do in the resolver level.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/ValidatableCommandClause.java (hunk 1 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/ValidatableCommandClause.java:34]: yes, I think so.

**Pull request merged**